pkg/handler/unrouted_handler.go

```diff
@@ -217,8 +217,15 @@ func (handler *UnroutedHandler) Middleware(h http.Handler) http.Handler {
 
 		header := w.Header()
 
-		if origin := r.Header.Get("Origin"); origin != "" {
+		var origin = handler.config.CorsOrigin
+		if origin == "" {
+			origin = r.Header.Get("Origin")
+		}
+
+		if origin != "" {
+
 			header.Set("Access-Control-Allow-Origin", origin)
+			header.Set("Vary", "Origin")
```

Comment:
This is just for correctness ( https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Access-Control-Allow-Origin#cors_and_caching )